x-pack/metricbeat/module/aws/ec2/_meta/data.json

```diff
@@ -39,18 +43,37 @@
         }
     },
     "cloud": {
-        "availability_zone": "eu-central-1b",
+        "availability_zone": "us-west-2a",
         "image": {
-            "id": "ami-556ef13a"
+            "id": "ami-01e24be29428c15b2"
         },
         "instance": {
-            "id": "i-08027f3dcddb9a976"
+            "core": {
+                "count": 1
+            },
+            "id": "i-077bdaf7e5d81bba3",
+            "monitoring": {
+                "state": "disabled"
+            },
+            "private": {
+                "dns_name": "ip-172-31-26-12.us-west-2.compute.internal",
+                "ip": "172.31.26.12"
+            },
+            "public": {
+                "dns_name": "ec2-34-217-213-210.us-west-2.compute.amazonaws.com",
+                "ip": "34.217.213.210"
+            },
+            "state": {
```

Comment:
I think it's a good idea to have the state info as part of each event. I'm not so sure if all the other state data should also be added all the time as it never (rarely) changes for the same instance. Can the values above be adjusted without getting a new id?

My main worry is that we generate quite a bit more data then needed. At the same time if it's always the same, it should compress well.

Reply:
@exekias WDYT?

Reply:
@ruflin Yeah we can definitely trim some data down here. I was more trying to show what are the information that might be useful. `monitoring state` can be changed without getting a new ID for sure.

Reply:
I mean the values except monitoring state :-) But I would assume also DNS can potentially change, not sure about IP.

Reply:
@exekias The reason why I added under `cloud.instance` is also because `cloud.instance.*` is ec2 specific. For example `cloud.instance.id`. Also all the metrics under `aws.ec2.*` are queried from AWS cloudwatch api. But metrics like `monitoring.state` and `state.code` are from ec2 api describeInstances. @ruflin What do you think?

Reply:
I agree with @exekias that for now I would only put the fields in ECS here: https://github.com/elastic/ecs#cloud